src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeRanges.cs

```diff
@@ -32,9 +32,8 @@ private static UnicodeRange CreateEmptyRange(ref UnicodeRange range)
         {
             // If the range hasn't been created, create it now.
             // It's ok if two threads race and one overwrites the other's 'range' value.
-            var newRange = new UnicodeRange(0, 0);
-            Volatile.Write(ref range, newRange);
-            return newRange;
+            var range = new UnicodeRange(0, 0);
+            return range;
```

Comment:
Yes, the reference won't get torn and ctor finishes before the object is visible. What else could cause racing threads to see partially instantiated object? I am not following; we are talking about very straightforward code that is all over the place in our code base. 

Also, not that the CLUI spec talks about this explicitly:

I.12.6.8 Other memory model i s sues
All memory allocated for static variables (other than those assigned RVAs within a PE file, see
Partition II) and objects shall be zeroed before they are made visible to any user code.
A conforming implementation of the CLI shall ensure that, even in a multi-threaded environment
and without proper user synchronization, objects are allocated in a manner that prevents
unauthorized memory access and prevents invalid operations from occurring. In particular, on
multiprocessor memory systems where explicit synchronization is required to ensure that all
relevant data structures are visible (for example, vtable pointers) the Execution Engine shall be
responsible for either enforcing this synchronization automatically or for converting errors due to
lack of synchronization into non-fatal, non-corrupting, user-visible exceptions.
It is explicitly not a requirement that a conforming implementation of the CLI guarantee that all
state updates performed within a constructor be uniformly visible before the constructor
completes. CIL generators can ensure this requirement themselves by inserting appropriate calls
to the memory barrier or volatile write instructions.